File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientReadEpisode.java
package com.mytlogos.enterprise.background.api.model;

import org.joda.time.DateTime;

public class ClientReadEpisode {
    private int episodeId;
    private DateTime readDate;
    private float progress;

    public ClientReadEpisode(int episodeId, DateTime readDate, float progress) {
        this.episodeId = episodeId;
        this.readDate = readDate;
        this.progress = progress;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ClientReadEpisode that = (ClientReadEpisode) o;

        return getEpisodeId() == that.getEpisodeId();
    }

    @Override
    public int hashCode() {
        return getEpisodeId();
    }

    @Override
    public String toString() {
        return "ClientReadEpisode{" +
                "id=" + episodeId +
                ", readDate=" + readDate +
                ", progress=" + progress +
                '}';
    }

    public int getEpisodeId() {
        return episodeId;
    }

    public DateTime getReadDate() {
        return readDate;
    }

    public float getProgress() {
        return progress;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/model/ClientReadEpisode.kt
package com.mytlogos.enterprise.background.api.model

import org.joda.time.DateTime

/**
 * API Model for ReadEpisode in [ClientUser].
 */
data class ClientReadEpisode(
    val episodeId: Int,
    val readDate: DateTime,
    val progress: Float,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ClientReadEpisode
        return episodeId == that.episodeId
    }

    override fun hashCode(): Int {
        return episodeId
    }
}